3.3.0 build warnings on Debian unstable

I'm working on building 3.3.0 on Debian unstable. Note that I'm building against the [Debian packaged version of googletest](https://tracker.debian.org/pkg/googletest).

```
...
The Meson build system
Version: 0.46.1
Source dir: /build/zimlib-3.3.0
Build dir: /build/zimlib-3.3.0/obj-x86_64-linux-gnu
Build type: native build
Project name: libzim
Native C compiler: cc (gcc 7.3.0 "cc (Debian 7.3.0-19) 7.3.0")
Appending CFLAGS from environment: '-g -O2 -fdebug-prefix-map=/build/zimlib-3.3.0=. -fstack-protector-strong -Wformat -Werror=format-security'
Appending LDFLAGS from environment: '-Wl,-z,relro'
Appending CPPFLAGS from environment: '-Wdate-time -D_FORTIFY_SOURCE=2'
Native C++ compiler: c++ (gcc 7.3.0 "c++ (Debian 7.3.0-19) 7.3.0")
Appending CXXFLAGS from environment: '-g -O2 -fdebug-prefix-map=/build/zimlib-3.3.0=. -fstack-protector-strong -Wformat -Werror=format-security'
Appending LDFLAGS from environment: '-Wl,-z,relro'
Appending CPPFLAGS from environment: '-Wdate-time -D_FORTIFY_SOURCE=2'
Build machine cpu family: x86_64
Build machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29)
Native dependency zlib found: YES 1.2.11
Native dependency liblzma found: YES 5.2.2
Native dependency xapian-core found: YES 1.4.5
Dependency threads found: YES
Native dependency icu-i18n found: YES 60.2
Configuring config.h using configuration
Dependency GTest found: YES (prebuilt)
Program cython3 found: NO
Dependency python3 found: NO
Build targets in project: 7
Found ninja-1.8.2 at /usr/bin/ninja
   dh_auto_build -O--buildsystem=meson
	cd obj-x86_64-linux-gnu && LC_ALL=C.UTF-8 ninja -j1 -v
...
[32/41] c++ -Itest/test@@cluster@exe -Itest -I../test -Iinclude -I../include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Werror -std=c++11 -g -O2 -fdebug-prefix-map=/build/zimlib-3.3.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -pthread  -MD -MQ 'test/test@@cluster@exe/cluster.cpp.o' -MF 'test/test@@cluster@exe/cluster.cpp.o.d' -o 'test/test@@cluster@exe/cluster.cpp.o' -c ../test/cluster.cpp
FAILED: test/test@@cluster@exe/cluster.cpp.o 
c++ -Itest/test@@cluster@exe -Itest -I../test -Iinclude -I../include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Werror -std=c++11 -g -O2 -fdebug-prefix-map=/build/zimlib-3.3.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -pthread  -MD -MQ 'test/test@@cluster@exe/cluster.cpp.o' -MF 'test/test@@cluster@exe/cluster.cpp.o.d' -o 'test/test@@cluster@exe/cluster.cpp.o' -c ../test/cluster.cpp
../test/cluster.cpp: In member function ‘virtual void {anonymous}::CluterTest_read_extended_cluster_Test::TestBody()’:
../test/cluster.cpp:290:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, &a, 1);
   ~~~~~^~~~~~~~~~~
../test/cluster.cpp:295:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, out_buf, sizeof(uint64_t));
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:299:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, out_buf, sizeof(uint64_t));
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:303:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, out_buf, sizeof(uint64_t));
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:307:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, out_buf, sizeof(uint64_t));
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:311:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, out_buf, sizeof(uint64_t));
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:313:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, blob0.c_str(), blob0.size());
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:314:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, blob1.c_str(), blob1.size());
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:315:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, blob2.c_str(), blob2.size());
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../test/cluster.cpp:319:8: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(fd, &a, 1);
   ~~~~~^~~~~~~~~~~
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
dh_auto_build: cd obj-x86_64-linux-gnu && LC_ALL=C.UTF-8 ninja -j1 -v returned exit code 1
```

Removing `werror=true` from meson.build makes the build complete, but is that OK as a long term solution?

--- a/test/cluster.cpp
+++ b/test/cluster.cpp
@@ -275,6 +275,7 @@ TEST(CluterTest, read_extended_cluster)
 {
   std::FILE* tmpfile = std::tmpfile();
   int fd = fileno(tmpfile);
+  ssize_t bytes_written;
 
   std::string blob0("123456789012345678901234567890");
   std::string blob1("ABCDEFGHIJKLMNOPQRSTUVWXYZ");
@@ -285,36 +286,42 @@ TEST(CluterTest, read_extended_cluster)
   zim::offset_type offset = 5*sizeof(uint64_t);
 
   char a = 0x11;
-  write(fd, &a, 1);
+  bytes_written = write(fd, &a, 1);
 
   char out_buf[sizeof(uint64_t)];
 
   zim::toLittleEndian(offset, out_buf);
-  write(fd, out_buf, sizeof(uint64_t));
+  bytes_written = write(fd, out_buf, sizeof(uint64_t));
 
   offset += blob0.size();
   zim::toLittleEndian(offset, out_buf);
-  write(fd, out_buf, sizeof(uint64_t));
+  bytes_written = write(fd, out_buf, sizeof(uint64_t));
 
   offset += blob1.size();
   zim::toLittleEndian(offset, out_buf);
-  write(fd, out_buf, sizeof(uint64_t));
+  bytes_written = write(fd, out_buf, sizeof(uint64_t));
 
   offset += blob2.size();
   zim::toLittleEndian(offset, out_buf);
-  write(fd, out_buf, sizeof(uint64_t));
+  bytes_written = write(fd, out_buf, sizeof(uint64_t));
 
   offset += bigger_than_4g;
   zim::toLittleEndian(offset, out_buf);
-  write(fd, out_buf, sizeof(uint64_t));
+  bytes_written = write(fd, out_buf, sizeof(uint64_t));
 
-  write(fd, blob0.c_str(), blob0.size());
-  write(fd, blob1.c_str(), blob1.size());
-  write(fd, blob2.c_str(), blob2.size());
+  bytes_written = write(fd, blob0.c_str(), blob0.size());
+  ASSERT_EQ(bytes_written, (ssize_t)blob0.size());
+
+  bytes_written = write(fd, blob1.c_str(), blob1.size());
+  ASSERT_EQ(bytes_written, (ssize_t)blob1.size());
+
+  bytes_written = write(fd, blob2.c_str(), blob2.size());
+  ASSERT_EQ(bytes_written, (ssize_t)blob2.size());
+
   lseek(fd , bigger_than_4g-1, SEEK_CUR);
 //  std::fseek(tmpfile, bigger_than_4g-1, SEEK_CUR);
   a = '\0';
-  write(fd, &a, 1);
+  bytes_written = write(fd, &a, 1);
   fflush(tmpfile);
 
   auto fileCompound = std::shared_ptr<zim::FileCompound>(new zim::FileCompound(tmpfile));

